### PR TITLE
feat: Add Default instance for Config

### PR DESCRIPTION
**Pull request by aloussase**
I think this would be nice for people who are already using data-default-class.

### CODE REVIEW

File: dotenv.cabal [hunk 1]
```diff
@@ -134,6 +135,7 @@ test-suite dotenv-test
                      , exceptions >= 0.8 && < 0.11
                      , hspec-megaparsec >= 2.0 && < 3.0
                      , mtl
+                     , data-default-class >= 0.1.2 && < 0.2
```
Comment (CristhianMotoche) [inline dotenv.cabal:138]: Is it necessary to add it to the test-suite?

**Review by CristhianMotoche — COMMENTED** [review]
Hey @aloussase Thanks a lot for your contribution. I like the idea to include this instance. I just left one question in your PR. Could you please take a look at it?

File: dotenv.cabal (hunk 1 above)
Comment (aloussase) [reply dotenv.cabal:138]: I think it is, because the test suite specifies the src directory` in `hs-source-dirs`.

Comment (CristhianMotoche) [reply dotenv.cabal:138]: Oh, you're right. I'll fix that in a separated PR. Thanks man!

**Review by CristhianMotoche — APPROVED** [review]
LGTM! Thank you @aloussase I'll try to get it to hackage as soon as I can.

File: dotenv.cabal (hunk 1 above)
Comment (aloussase) [reply dotenv.cabal:138]: I thought it was on purpose haha you're welcome :)

**Pull request merged**